evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -75,16 +75,28 @@ def __init__(
             fill_value=categorical_fill_value,
             **kwargs,
         )
-        self._boolean_imputer = SimpleImputer(
-            impute_strategy=boolean_impute_strategy,
-            fill_value=boolean_fill_value,
-            **kwargs,
-        )
-        self._numeric_imputer = SimpleImputer(
-            impute_strategy=numeric_impute_strategy,
-            fill_value=numeric_fill_value,
-            **kwargs,
-        )
+        if boolean_impute_strategy == "knn":
+            self._boolean_imputer = KNNImputer(
+                number_neighbors=1,
```

Comment:
why is `number_neighbors=1` here but `number_neighbors=3` for numeric columns?

Reply:
If we're dealing with booleans, if I do any number of neighbors above 1, KNNImputer imputes the boolean as numbers such as 0.3333 or 0.5555.


I could make it so that it rounds to 0 (False) if the imputed number < 0.5 and round to 1 (True) if the imputed number >= 0.5

Reply:
I think thats a good idea - we should be able to up the number of neighbors. Can you file an issue for this?